src/contract/types.rs

```diff
@@ -35,70 +42,140 @@ pub fn encode_document_field_type(
         }
         DocumentFieldType::Integer => {
             let value_as_integer = value.as_integer().ok_or(field_type_match_error)?;
-            let value_as_u64: u64 = value_as_integer
+            let value_as_i64: i64 = value_as_integer
                 .try_into()
                 .map_err(|_| Error::CorruptedData(String::from("expected integer value")))?;
-            Ok(Some(value_as_u64.to_be_bytes().to_vec()))
+
+            encode_integer(value_as_i64)
         }
         DocumentFieldType::Float => {
             let value_as_float = value.as_float().ok_or(field_type_match_error)?;
             Ok(Some(value_as_float.to_be_bytes().to_vec()))
         }
+        DocumentFieldType::ByteArray => {
+            // Byte array could either be raw bytes or encoded as a base64 string
+            if value.is_text() {
```

Comment:
ByteArray actually always will be raw bytes

Reply:
Currently we test by converting json to cbor and byte arrays are represented as base64 in the json files, it's temporary tho will eventually move the base64 decoding to conversion function.

Reply:
yes @shumkov, this is basically just to allow tests to be much easier.

Reply:
Ok, sounds good. Should we add a TODO there to do not forget?

Reply:
I'm not sure we need to, what's the point of not keeping this?